sdk/core/azure-core/tests/test_universal_pipeline.py

```diff
@@ -408,3 +414,72 @@ def emit(self, record):
     assert mock_handler.messages[4].message == 'Response headers:'
 
     mock_handler.reset()
+
+def test_retry_seekable_stream():
+    class MockTransport(HttpTransport):
+        def __init__(self):
+            self._first = True
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            pass
+        def close(self):
+            pass
+        def open(self):
+            pass
+
+        def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineResponse
+            if self._first:
+                self._first = False
+                request.body.seek(0,2)
+                raise AzureError('fail on first')
+            position = request.body.tell()
+            assert not position
```

Comment:
Detail: `tell` is actually returning an index as integer, so for clarity here I would have written "assert position == 0" since there is no boolean logic in `tell` but I agree that technically the assert does the same thing, but as a side-effect of 0 being False.